app/src/main/java/org/breezyweather/main/MainActivity.kt

```diff
@@ -327,8 +334,7 @@ class MainActivity : GeoActivity(),
                         }
 
                         if (showLocationPermissionDialog && !viewModel.statementManager.isLocationPermissionDialogAlreadyShown) {
```

Comment:
Instead of dismissing the dialog in `onPause`, wouldn't it be easier to add a new condition `dialogPerLocationAlert == null` in the `if` statement, so the dialog is only created once?

Reply:
Unfortunately that doesn't seem to work when I test it. If the user pauses the app while the dialog is open and then resumes it, the dialog is bypassed and it jumps to requesting permissions. As far as I can tell it's because the activity is being recreated, but when the activity is *recreated* rather than created, that variable would no longer be null.

Reply:
You're right. A nested `if` expression with that condition should work though.

```
if (showLocationPermissionDialog && !viewModel.statementManager.isLocationPermissionDialogAlreadyShown) {
    if (dialogPerLocationAlert == null) {
        ...
```

Reply:
That does work. My only concern is that this method causes a memory leak when the orientation changes or the app is closed. Dismissing the dialog seems to avoid that issue.

Reply:
Thanks for pointing that out. That also seems to be an issue in the current release. Your solution with dismissing the dialog does indeed solve that 👍 

While having a closer look at this, I also noticed that a similar issue occurs when the dialog for the background location permission is [shown](https://github.com/breezy-weather/breezy-weather/blob/dc1808121702f1bbc9cba527290b123232451fe8/app/src/main/java/org/breezyweather/main/MainActivity.kt#L536). As it's not directly related to #216, I'll probably just open a separate PR to fix this.